packages/react/src/Accordion/Accordion.tsx

```diff
@@ -1,146 +1,105 @@
+const useAccordionContext = (): AccordionContextType => {
+  const context = useContext(AccordionContext)
 
-type AccordionHeadingProps = PropsWithChildren<BaseProps<HTMLHeadingElement>> &
-  React.HTMLAttributes<HTMLDetailsElement> & {
-    className?: string
-    as?: 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
-    reversedToggles?: boolean
-    variant?: 'default' | 'emphasis'
-    open?: boolean // private prop passed from AccordionRoot
-    handleOpen?: (boolean) => void // private prop passed from AccordionRoot
-    parentRef?: React.RefObject<HTMLDetailsElement> // private prop passed from AccordionRoot
-    toggleColor?: (typeof AccordionToggleColors)[number]
+  if (context === null) {
+    throw new Error('Unable to find Accordion provider. Did you forget to wrap your component in an Accordion?')
   }
 
-export const AccordionHeading = forwardRef<HTMLHeadingElement, AccordionHeadingProps>(
+  return context
+}
+
+export const AccordionRoot = forwardRef<HTMLDetailsElement, AccordionRootProps>(
   (
-    {
-      children,
-      className,
-      as = 'h4',
-      variant = 'default',
-      toggleColor,
-      reversedToggles,
-      open,
-      handleOpen,
-      parentRef,
-      style,
-      ...rest
-    },
-    ref,
+    {className, variant = 'default', open, onToggle: _onToggle, onKeyDown: _onKeyDown, handleOpen, ...rest},
+    forwardedRef,
   ) => {
-    const handleClick = useCallback(
-      (event: React.MouseEvent<HTMLDetailsElement, MouseEvent>) => {
-        event.preventDefault()
+    const ref = useProvidedRefOrCreate(forwardedRef as RefObject<HTMLDetailsElement>)
+    const accordionContextValue = useMemo(() => ({variant}), [variant])
 
-        if (handleOpen) {
-          handleOpen(!open)
-        }
+    const onToggle = useCallback<NonNullable<AccordionRootProps['onToggle']>>(
+      event => {
+        _onToggle?.(event)
+        handleOpen?.(event.currentTarget.open)
       },
-      [handleOpen, open],
+      [_onToggle, handleOpen],
     )
 
-    const handleKeyPress = useCallback(
-      (event: React.KeyboardEvent<HTMLDetailsElement>) => {
-        if (event.key === 'Enter' || event.key === ' ') {
-          event.preventDefault()
-          if (handleOpen) handleOpen(!open)
-        }
+    const onKeyDown = useCallback<NonNullable<AccordionRootProps['onKeyDown']>>(
+      event => {
+        _onKeyDown?.(event)
+
+        const details = ref.current
 
-        if (event.key === 'Escape') {
-          event.preventDefault()
-          if (handleOpen) handleOpen(false)
+        if (event.key === 'Escape' && details?.open) {
+          details.open = false
+          details.querySelector('summary')?.focus()
         }
       },
-      [handleOpen, open],
+      [_onKeyDown, ref],
     )
 
+    return (
+      <AccordionContext.Provider value={accordionContextValue}>
+        {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
```

Comment:
We try to avoid using eslint disable rules starting with `jsx-a11y/*` because they are tracked and reported on externally. Can we remove this?

Reply:
👍 5b3a8e1e8b8c12819b7273266d37b9f15f64823a